server/src/main/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdSettings.java

```diff
@@ -39,6 +39,20 @@ public class DiskThresholdSettings {
         Setting.Property.Dynamic,
         Setting.Property.NodeScope
     );
+    public static final Setting<ByteSizeValue> CLUSTER_ROUTING_ALLOCATION_LOW_DISK_MAX_HEADROOM_SETTING = new Setting<>(
+        "cluster.routing.allocation.disk.watermark.low.max_headroom",
+        (settings) -> {
+            if (CLUSTER_ROUTING_ALLOCATION_LOW_DISK_WATERMARK_SETTING.exists(settings)) {
+                return "-1";
+            } else {
+                return "150GB";
+            }
+        },
+        (s) -> ByteSizeValue.parseBytesSizeValue(s, "cluster.routing.allocation.disk.watermark.low.max_headroom"),
+        new MaxHeadroomValidator(),
```

Comment:
Should we also only permit these values when using proportional watermarks, not absolute ones? I think it could be confusing to accept headroom values and absolute watermarks.

Reply:
For the moment, if the watermarks are set to absolute values, the max headroom values are not used even if they are set. This is also explained in the documentation. But I do see that it can be confusing if both are set to absolute values. However, I am not sure it is a cause for an error/exception. Maybe it would make sense to log a warning message? If you agree, maybe I could add it in MaxHeadroomValidator?

Reply:
Yes I think we should reject this combination of settings as invalid. It's generally better to be strict about this sort of thing up front rather than having to deal with confused users later on.

Reply:
Done in latest commit. Please check and unresolve conversation if not sufficient.